File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/MapsPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record MapsPacket(int itemDamage, byte[] data) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeVarInt(os, itemDamage);
        os.writeShort(data.length);
        os.writeBytes(data);
    }

    @Override
    public byte getId() {
        return 0x34;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/MapsPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class MapsPacket(val itemDamage: Int, val data: ByteArray) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        writeVarInt(os, itemDamage)
        os.writeShort(data.size)
        os.writeBytes(data)
    }

    override fun getId(): Byte {
        return 0x34
    }
}
